osu.Game/Modes/UI/HitRenderer.cs

```diff
@@ -60,7 +61,12 @@ public abstract class HitRenderer<TObject> : HitRenderer
 
         protected HitRenderer(WorkingBeatmap beatmap)
         {
+            Debug.Assert(beatmap != null, "HitRenderer initialized with a null beatmap.");
+
+            // Convert + process the beatmap
             Beatmap = CreateBeatmapConverter().Convert(beatmap.Beatmap);
+            Beatmap.HitObjects.ForEach(h => CreateBeatmapProcessor().SetDefaults(h, Beatmap));

```

Comment:
I agree with all of this, and I have a separate branch ready to work on defaults, because I want to put a bit more thought into it.